From dfac09e5eb3525ecc93c4cd0a42f97573ec200d3 Mon Sep 17 00:00:00 2001
From: Jason Hung <jhung@globalscaletechnologies.com>
Date: Tue, 10 Sep 2024 12:34:35 +0800
Subject: [PATCH] board: mt8370: add gti-mt510ss-evk firmware update uuid

add firmware update uuid for globalscale-mt510ss-evk platform
---
 board/mediatek/mt8370/mt8370_evk.c | 26 +++++++++++++++++++++++++-
 include/configs/mt8370.h           | 16 ++++++++++++++++
 2 files changed, 41 insertions(+), 1 deletion(-)

diff --git a/board/mediatek/mt8370/mt8370_evk.c b/board/mediatek/mt8370/mt8370_evk.c
index 889e8dce6a..49763aab10 100644
--- a/board/mediatek/mt8370/mt8370_evk.c
+++ b/board/mediatek/mt8370/mt8370_evk.c
@@ -47,6 +47,12 @@ static bool board_is_genio_510_evk(void)
 		of_machine_is_compatible("mediatek,genio-510-evk");
 }
 
+static bool board_is_gti_mt510ss_evk(void)
+{
+	return CONFIG_IS_ENABLED(TARGET_MT8370) &&
+		of_machine_is_compatible("globalscale,mt510ss-evk");
+}
+
 void mediatek_capsule_update_board_setup(void)
 {
 	fw_images[0].image_index = MT8370_FIT_IMAGE;
@@ -55,7 +61,25 @@ void mediatek_capsule_update_board_setup(void)
 	fw_images[3].image_index = MT8370_FW_IMAGE;
 	fw_images[4].image_index = MT8370_ENV_IMAGE;
 
-	if (board_is_genio_510_evk()) {
+	if (board_is_gti_mt510ss_evk()) {
+		efi_guid_t image_type_guid = GTI_MT510SS_EVK_FIT_IMAGE_GUID;
+		efi_guid_t uboot_image_type_guid = GTI_MT510SS_EVK_FIP_IMAGE_GUID;
+		efi_guid_t bl2_image_type_guid = GTI_MT510SS_EVK_BL2_IMAGE_GUID;
+		efi_guid_t fw_image_type_guid = GTI_MT510SS_EVK_FW_IMAGE_GUID;
+		efi_guid_t env_image_type_guid = GTI_MT510SS_EVK_ENV_IMAGE_GUID;
+
+		guidcpy(&fw_images[0].image_type_id, &image_type_guid);
+		guidcpy(&fw_images[1].image_type_id, &uboot_image_type_guid);
+		guidcpy(&fw_images[2].image_type_id, &bl2_image_type_guid);
+		guidcpy(&fw_images[3].image_type_id, &fw_image_type_guid);
+		guidcpy(&fw_images[4].image_type_id, &env_image_type_guid);
+
+		fw_images[0].fw_name = u"GTI-MT510SS-EVK-FIT";
+		fw_images[1].fw_name = u"GTI-MT510SS-EVK-FIP";
+		fw_images[2].fw_name = u"GTI-MT510SS-EVK-BL2";
+		fw_images[3].fw_name = u"GTI-MT510SS-EVK-FW";
+		fw_images[4].fw_name = u"GTI-MT510SS-EVK-ENV";
+	} else if (board_is_genio_510_evk()) {
 		efi_guid_t image_type_guid = GENIO_510_EVK_FIT_IMAGE_GUID;
 		efi_guid_t uboot_image_type_guid = GENIO_510_EVK_FIP_IMAGE_GUID;
 		efi_guid_t bl2_image_type_guid = GENIO_510_EVK_BL2_IMAGE_GUID;
diff --git a/include/configs/mt8370.h b/include/configs/mt8370.h
index 3266387523..d92f189629 100644
--- a/include/configs/mt8370.h
+++ b/include/configs/mt8370.h
@@ -33,6 +33,22 @@
 	EFI_GUID(0xf967dfc1, 0xbb75, 0x4439, 0x98, 0x8f, \
 		 0xab, 0xe6, 0xc0, 0xa2, 0xd0, 0x19)
 
+#define GTI_MT510SS_EVK_FIT_IMAGE_GUID \
+	EFI_GUID(0x8b37443b, 0xc596, 0x45fd, 0x91, 0x3c, \
+		0x0a, 0xdd, 0xc7, 0x6f, 0x29, 0xc6)
+#define GTI_MT510SS_EVK_FIP_IMAGE_GUID \
+	EFI_GUID(0xf8191c07, 0x8c6f, 0x4f10, 0x96, 0x55, \
+		0x1b, 0x85, 0xed, 0xc7, 0x63, 0xe7)
+#define GTI_MT510SS_EVK_BL2_IMAGE_GUID \
+	EFI_GUID(0x87ef4432, 0xe654, 0x4ee5, 0x8e, 0xc7, \
+		0x11, 0x63, 0x8d, 0x9f, 0x27, 0x3f)
+#define GTI_MT510SS_EVK_FW_IMAGE_GUID \
+	EFI_GUID(0x531a7197, 0xb7cf, 0x4e25, 0x8f, 0x45, \
+		0xe8, 0x11, 0x02, 0xb7, 0xaa, 0xa6)
+#define GTI_MT510SS_EVK_ENV_IMAGE_GUID \
+	EFI_GUID(0x0826fd4e, 0x6eda, 0x414c, 0xa1, 0x81, \
+		0xfe, 0xbb, 0x02, 0xd3, 0x58, 0xa9)
+
 /* Environment settings */
 #include <config_distro_bootcmd.h>
 
-- 
2.34.1

